src/groupeddataframe/splitapplycombine.jl

```diff
@@ -979,8 +1110,15 @@ end
 function _combine(fun::Base.Callable, gd::GroupedDataFrame, ::Nothing)
     firstres = fun(gd[1])
     firstmulticol = firstres isa MULTI_COLS_TYPE
+    if !(firstres isa Union{AbstractVecOrMat, AbstractDataFrame,
```

Comment:
Would there be a way to avoid repeating this in each method?

Reply:
No - this is a part of optimization you wanted (in the earlier comment as TODO for performance): we do not want to calculate `idx_agg` each time as single-row function is encountered, so we have to compute it before we call `_combine_with_first` exactly once if it is needed to be computed.

Reply:
Sorry, my question was about whether it would be possible to reorganize the code to reduce duplication (not about performance).

Reply:
OK - I have added an intermediate function that captures duplicate code (essentially it handles combining when returning multiple columns is allowed).